File: scripts/generate-icons.js
/**
 * Script para generar íconos PWA faltantes desde íconos existentes
 * Requiere: npm install sharp
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

// Ruta base de los íconos
const ICONS_PATH = path.join(__dirname, '..', 'public', 'assets', 'icons');

// Configuración de íconos a generar
// formato: [archivo_fuente, tamaño_salida, nombre_salida]
const ICONS_TO_GENERATE = [
    ['web-app-manifest-192x192.png', 144, 'icon-144x144.png'],
    ['web-app-manifest-192x192.png', 128, 'icon-128x128.png'],
    ['favicon-96x96.png', 72, 'icon-72x72.png'],
    ['favicon-96x96.png', 48, 'icon-48x48.png'],
];

async function generateIcon(sourceFile, targetSize, outputFile) {
    const sourcePath = path.join(ICONS_PATH, sourceFile);
    const outputPath = path.join(ICONS_PATH, outputFile);
    
    try {
        // Verificar que existe el archivo fuente
        if (!fs.existsSync(sourcePath)) {
            console.log(`❌ Error: No se encontró ${sourceFile}`);
            return false;
        }

        // Redimensionar imagen
        await sharp(sourcePath)
            .resize(targetSize, targetSize, {
                fit: 'cover',
                position: 'center'
            })
            .png({ quality: 100, compressionLevel: 9 })
            .toFile(outputPath);
        
        console.log(`✅ Generado: ${outputFile} (${targetSize}x${targetSize})`);
        return true;
        
    } catch (error) {
        console.log(`❌ Error generando ${outputFile}: ${error.message}`);
        return false;
    }
}

async function main() {
    console.log('🎨 Generador de Íconos PWA - SENAttend\n');
    
    // Verificar que existe la carpeta de íconos
    if (!fs.existsSync(ICONS_PATH)) {
        console.log(`❌ Error: No se encuentra la carpeta ${ICONS_PATH}`);
        console.log('   Asegúrate de ejecutar este script desde la carpeta scripts/');
        return;
    }
    
    // Generar cada ícono
    let successCount = 0;
    for (const [source, size, output] of ICONS_TO_GENERATE) {
        const success = await generateIcon(source, size, output);
        if (success) successCount++;
    }
    
    console.log(`\n✨ Completado: ${successCount}/${ICONS_TO_GENERATE.length} íconos generados`);
    
    if (successCount === ICONS_TO_GENERATE.length) {
        console.log('\n📝 Nota: Actualiza el manifest.json para incluir los nuevos íconos.');
    } else {
        console.log('\n⚠️ Algunos íconos no se pudieron generar. Revisa los errores arriba.');
    }
}

// Ejecutar
main().catch(error => {
    console.error('\n❌ Error inesperado:', error.message);
    process.exit(1);
});

File: scripts/generate-icons.py
"""
Script para generar íconos PWA faltantes desde íconos existentes
Requiere: pip install Pillow
"""

from PIL import Image
import os

# Ruta base de los íconos
ICONS_PATH = os.path.join('public', 'assets', 'icons')

# Configuración de íconos a generar
# formato: (archivo_fuente, tamaño_salida, nombre_salida)
ICONS_TO_GENERATE = [
    ('web-app-manifest-192x192.png', 144, 'icon-144x144.png'),
    ('web-app-manifest-192x192.png', 128, 'icon-128x128.png'),
    ('favicon-96x96.png', 72, 'icon-72x72.png'),
    ('favicon-96x96.png', 48, 'icon-48x48.png'),
]

def generate_icon(source_file, target_size, output_file):
    """Genera un ícono redimensionado"""
    source_path = os.path.join(ICONS_PATH, source_file)
    output_path = os.path.join(ICONS_PATH, output_file)
    
    try:
        # Abrir imagen fuente
        with Image.open(source_path) as img:
            # Convertir a RGBA si no lo está
            if img.mode != 'RGBA':
                img = img.convert('RGBA')
            
            # Redimensionar con alta calidad
            resized = img.resize((target_size, target_size), Image.Resampling.LANCZOS)
            
            # Guardar
            resized.save(output_path, 'PNG', optimize=True)
            print(f'✅ Generado: {output_file} ({target_size}x{target_size})')
            return True
            
    except FileNotFoundError:
        print(f'❌ Error: No se encontró {source_file}')
        return False
    except Exception as e:
        print(f'❌ Error generando {output_file}: {str(e)}')
        return False

def main():
    print('🎨 Generador de Íconos PWA - SENAttend\n')
    
    # Verificar que existe la carpeta de íconos
    if not os.path.exists(ICONS_PATH):
        print(f'❌ Error: No se encuentra la carpeta {ICONS_PATH}')
        print('   Asegúrate de ejecutar este script desde la raíz del proyecto.')
        return
    
    # Generar cada ícono
    success_count = 0
    for source, size, output in ICONS_TO_GENERATE:
        if generate_icon(source, size, output):
            success_count += 1
    
    print(f'\n✨ Completado: {success_count}/{len(ICONS_TO_GENERATE)} íconos generados')
    
    if success_count == len(ICONS_TO_GENERATE):
        print('\n📝 Nota: Actualiza el manifest.json para incluir los nuevos íconos.')
    else:
        print('\n⚠️ Algunos íconos no se pudieron generar. Revisa los errores arriba.')

if __name__ == '__main__':
    try:
        main()
    except KeyboardInterrupt:
        print('\n\n⏹️ Operación cancelada por el usuario.')
    except Exception as e:
        print(f'\n❌ Error inesperado: {str(e)}')
